### DISCUSSION TITLE
Option to rewrite all URLs in content to point to archived versions

### DISCUSSION THREAD

**Discussion by superpupervlad** [ArchiveBox/ArchiveBox#1395] [Q&A]
For example, I save page example.com to webarchive. And this page has link to example.com/some_url.
When I will click on this link from https://web.archive.org/web/123/https://example.com the link won't lead me to example.com/some_url but to https://web.archive.org/web/123/https://example.com/some_url.
Is there an option (or plans to add this option) in archivebox to achieve something similar?

**Comment by pirate** [comment]
This is known as "URL Rewriting" and it's traditionally done in one of three ways:
- at archive time before saving to disk (not ideal, but easy, see `wget` and `SiteSucker`)
- at replay time on the server/proxy before it gets served to a client (difficult but doable, see `pywb`)
- at replay time in the browser of the client viewing the archive (see `wayback-everywhere` and `wombat.js`, easiest and lightest weight option)

There are no plans to add any of these to ArchiveBox at the moment, but if someone wants to build a Chrome Extension to do client-side rewriting at replay time, I'd be down to add a link to our README/add it to our Github org and help maintain it.

Check out these alternative solutions for now:

- `pywb` (python app that does proxy-replay / server-side rewriting, or client-side rewriting with JS)
  - https://github.com/ikreymer/pywb-proxy-demo
  - https://pywb.readthedocs.io/en/latest/manual/configuring.html#proxy-mode-rewriting ⭐️
  - https://pywb.readthedocs.io/en/latest/manual/rewriter.html#rewriter
  - https://pywb.readthedocs.io/en/latest/manual/configuring.html#wombat
- `wayback-everywhere` (extension that does client-side rewriting)
  - https://chromewebstore.google.com/detail/wayback-everywhere/heienainncnjjejmamdkmbbfmblpffjj
  *Someone should fork this extension to develop an equivalent for ArchiveBox!*
- SiteSucker (macOS app that does archive-time rewriting before saving html to disk)
  - https://ricks-apps.com/osx/sitesucker/index.html
  - https://ricks-apps.com/osx/sitesucker/archive/2.x/2.9.x/2.9/manuals/en/pgs/FAQ.html#:~:text=Modification%20option%20to-,Localize,-.%20With%20this%20setting

**Reply by BinaryVixen899** [reply]
Just to make sure I'm understanding this correctly, this is by design behavior _for any circumstance?_
IE: I am not uploading to an archive (yet) but I have been using `--depth=1` to recursively archive things. With a filter, it works great, but when I examine the WARC files I notice that all of the links on the page point to the actual site, not to the other archived files. Am I doing something wrong or is this how the program works? (BTW, `ArchiveBox` is awesome and thank you for your work on it!) 

UPDATE: It looks like I may be able to get around this via merging the WARC files, at least when viewing a WARC via replayweb.page

**Reply by pirate** [reply]
This is how it works currently, ArchiveBox does not do any rewriting of URLs to point archives to other archives. It just snapshots each page separately and tries to preserve it as it was originally served as closely as possible.
